be/src/agent/heartbeat_server.cpp

```diff
@@ -118,6 +121,19 @@ void HeartbeatServer::heartbeat(THeartbeatResult& heartbeat_result, const TMaste
     }
 
     static auto num_hardware_cores = static_cast<int32_t>(CpuInfo::num_cores());
+
+#ifdef USE_STAROS
+    if (master_info.run_mode == TRunMode::SHARED_DATA && reboot_time == 0) {
+        auto update_config = UpdateConfigAction::instance();
+        if (update_config == nullptr) {
+            LOG(WARNING) << "update flush thread number ignored: UpdateConfigAction is not inited";
+        } else {
+            auto s = update_config->update_config("flush_thread_num_per_store", std::to_string(2 * num_hardware_cores));
```

Comment:
Doing so will cause the user-set parameters to fail each time the process is restarted and receives a heartbeat